src/runtime_src/core/tools/xbutil2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,134 @@
+             "Device and host configuration")
+{
+  const std::string longDescription = "Device and host configuration.";
+  setLongDescription(longDescription);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+
+void
+SubCmdConfigure::execute(const SubCmdOptions& _options) const
+{
+  // -- Common top level options ---
+  bool help = false;
+
+  po::options_description commonOptions("Common Options"); 
+  commonOptions.add_options()
+    ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+
+  po::options_description hiddenOptions("Hidden Options"); 
+
+  // -- Define the supporting option options ----
+  SubOptionOptions subOptionOptions;
+  subOptionOptions.emplace_back(std::make_shared<OO_HostMem>("host-mem"));
+
+  for (auto & subOO : subOptionOptions) {
+    if (subOO->isHidden()) 
+      hiddenOptions.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    else
+      commonOptions.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    subOO->setExecutable(getExecutableName());
+    subOO->setCommand(getName());
+  }
+
+  po::options_description allOptions("All Options");
+  allOptions.add(commonOptions);
+  allOptions.add(hiddenOptions);
+
+  // =========== Process the options ========================================
+
+  // 1) Process the common top level options 
+  po::parsed_options parsedCommonTop = 
+    po::command_line_parser(_options).
+    options(allOptions).          
+    allow_unregistered().           // Allow for unregistered options
+    run();                          // Parse the options
+
+  po::variables_map vm;
+
+  try {
+    po::store(parsedCommonTop, vm);  // Can throw
+    po::notify(vm);                  // Can throw (but really isn't used)
+
+    // Multual DRC
+    for (unsigned int index1 = 0; index1 < subOptionOptions.size(); ++index1) {
+      for (unsigned int index2 = index1 + 1; index2 < subOptionOptions.size(); ++index2) {
+        conflictingOptions(vm, subOptionOptions[index1]->longName(), subOptionOptions[index2]->longName());
+      }
+    }
+  } catch (const std::exception & e) {
+    std::cerr << "ERROR: " << e.what() << std::endl;
+    printHelp(commonOptions, hiddenOptions, subOptionOptions);
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  // Find the subOption;
+  std::shared_ptr<OptionOptions> optionOption;
+  for (auto& subOO : subOptionOptions) {
+    if (vm.count(subOO->longName().c_str()) != 0) {
+      optionOption = subOO;
+      break;
+    }
+  }
+
+  // No suboption print help
+  if (!optionOption) {
+    printHelp(commonOptions, hiddenOptions, subOptionOptions);
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  // 2) Process the top level options
+  std::vector<std::string> topOptions = po::collect_unrecognized(parsedCommonTop.options, po::include_positional);
+  if (help) {
```

Comment:
The curly brackets (e.g., '{' and '}') should not be used for 1 line operations.  Yeah, I know that I have done so in the past, but Soren has convinced me that we should do so.